src/wp-includes/blocks/index.php

```diff
@@ -12,6 +12,82 @@
 require BLOCKS_PATH . 'widget-group.php';
 require BLOCKS_PATH . 'require-dynamic-blocks.php';
 
+/**
+ * Registers core block style handles.
+ *
+ * While {@see register_block_style_handle()} is typically used for that, the way it is
+ * implemented is inefficient for core block styles. Registering those style handles here
+ * avoids unnecessary logic and filesystem lookups in the other function.
+ *
+ * @since 6.3.0
+ */
+function register_core_block_style_handles() {
+	if ( ! wp_should_load_separate_core_block_assets() ) {
+		return;
+	}
+
+	static $core_blocks_meta;
+	if ( ! $core_blocks_meta ) {
+		$core_blocks_meta = require ABSPATH . WPINC . '/blocks/blocks-json.php';
+	}
+
+	$includes_url   = includes_url();
+	$includes_path  = ABSPATH . WPINC . '/';
+	$suffix         = wp_scripts_get_suffix();
+	$wp_styles      = wp_styles();
+	$style_fields   = array(
+		'style'       => 'style',
+		'editorStyle' => 'editor',
+	);
+	$transient_name = 'wp_core_block_styles';
+	$files          = get_transient( $transient_name );
+	if ( ! $files ) {
+		$files = glob( __DIR__ . "/**/**{$suffix}.css" );
+		set_transient( $transient_name, $files );
```

Comment:
@felixarntz Should we disable this cache is the WP_DEBUG is enabled?

Reply:
@spacedmonkey Sounds good. Please add the same `todo` comment that is present on all the other places where we do caching around e.g. the `theme.json` stuff.

Reply:
@felixarntz Pushed the change.